scalding-core/src/main/scala/com/twitter/scalding/Job.scala

```diff
@@ -498,6 +498,29 @@ class ScriptJob(cmds: Iterable[String]) extends Job(Args("")) {
   }
 }
 
+/**
+ * Allows custom counter verification logic when the job completes.
+ */
+trait CounterVerification extends Job {
+  /**
+   * Verify counter values. The job will fail if this returns false or throws an exception.
+   */
+  def verifyCounters(counters: Map[StatKey, Long]): Boolean
```

Comment:
I was trying to keep this interface simple. The user can include message with `assert(_, msg)` or throw exception directly. Does that make sense?